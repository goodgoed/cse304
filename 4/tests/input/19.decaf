class test005
{
    public static void main()
    {   
        boolean a;
        a = 17 > 14;
        a = true || false;
        a = true && false;
        a = 17 != 14;
    }
}
