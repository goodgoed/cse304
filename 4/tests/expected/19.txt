--------------------------------------------------------------------------
Class Name: test005
Superclass Name: 
Fields:
Constructors:
Methods:
METHOD: 1, main, test005, public, static, void
Method Parameters: 
Variable Table:
VARIABLE: 1, a, local, boolean
Method Body:
Block([
Expr(Assign(Variable(1), Binary(gt, Constant(Integer-constant(17)), Constant(Integer-constant(14))), boolean, boolean)),
Expr(Assign(Variable(1), Binary(or, Constant(True), Constant(False)), boolean, boolean)),
Expr(Assign(Variable(1), Binary(and, Constant(True), Constant(False)), boolean, boolean)),
Expr(Assign(Variable(1), Binary(neq, Constant(Integer-constant(17)), Constant(Integer-constant(14))), boolean, boolean))
])
--------------------------------------------------------------------------
